samples/sample/src/main/java/com/microsoft/jfr/App.java

```diff
@@ -0,0 +1,67 @@
+package com.microsoft.jfr;
+
+import javax.management.InstanceNotFoundException;
+import javax.management.MBeanServerConnection;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.math.BigInteger;
+import java.nio.file.Paths;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.RecursiveAction;
+
+public class App 
+{
+    private static final ForkJoinPool threadPool = new ForkJoinPool();
+    private static final int range = 100000;
+
+    static class Fibonacci extends RecursiveAction {
+
+        public BigInteger NthFibonacci(int n) {
+            BigInteger[] fibonacciNumbers = new BigInteger[n+2];
```

Comment:
If this array was taken from some cache and that cache was static in the outer class, then we could drive up some memory usage and possibly create some memory churn. The cache could be a simple queue where NthFib takes the first element for its computation then returns it when the computation is done. Every now and then, the dequeue could return new memory, creating a slow leak.

Reply:
Have reimplemented the logic here. We now have a static cache array and we keep re-using it for every function call.